mmv1/third_party/terraform/services/kms/data_source_google_kms_crypto_key_versions_test.go

```diff
@@ -0,0 +1,37 @@
+package kms_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+)
+
+func TestAccDataSourceGoogleKmsCryptoKeyVersions_basic(t *testing.T) {
+	asymSignKey := acctest.BootstrapKMSKeyWithPurpose(t, "ASYMMETRIC_SIGN")
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceGoogleKmsCryptoKeyVersions_basic(asymSignKey.CryptoKey.Name),
+				Check:  resource.TestCheckResourceAttr("data.google_kms_crypto_key_versions.versions", "versions.#", "2"),
```

Comment:
Can you set up a test where we would expect a different number of versions to exist than the default? Maybe by disabling a key and then making sure it doesn't come back

Reply:
I added more test cases that involve the use of filter so that we can get different amount of versions (whether it's based on name or `state:enabled`)